src/zimscraperlib/zim/creator.py

```diff
@@ -225,15 +228,23 @@ def start(self):
     def validate_metadata(
         self,
         name: str,
-        value: bytes | str | datetime.datetime | datetime.date | Iterable[str],
+        value: (
+            int
```

Comment:
Why all those new types? We dont support them, there is no standard metadata that matches them and it needs to be stored as bytes/str anyway

Reply:
Because this is what is supported by `add_metadata`. Typically to silently convert from one type to str/bytes silently. And this is what is passed to the tests in `test_validate_metadata`.

However I agree that validate_metadata should check that after these automated conversions in `add_metadata` we have only bytes|str left. Is this what you intended?

Reply:
> Because this is what is supported by add_metadata

`add_metadata` in libzim? I suppose you mean `config_metadata` here… but you added those changes 😅
Do we want to silently convert from those types? Resolve if you think so

Reply:
I've significantly revisited all these methods around metadata in last commit https://github.com/openzim/python-scraperlib/pull/178/commits/75eed4e724979d38234033a930788c06f85c8a2f ; I expect it is now both clearer and easier to maintain. Only `config_metadata` and `config_dev_metadata` accepts many value types. `add_metadata` and `validate_metadata` only accepts `str | bytes` now.